edx_sysadmin/utils/utility.py

```diff
@@ -0,0 +1,23 @@
+"""
+Utility function defined here.
+"""
+
+import six
+from django.http import Http404
+from xmodule.modulestore.django import modulestore
+
+
+def get_course_by_id(course_key, depth=0):
```

Comment:
Shouldn't we just import this function from https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/courses.py#L94 rather than re-defining?

Reply:
Good question. I was thinking the same before but we should try to minimize the rely on edx methods. As we are already importing module store then I was thinking to build our own method for getting courses instead relying again on OpenEdx Method.

Reply:
I think we should better rely on edX's methods instead of getting something directly from modulestore because whenever some change will be done in modulestore edX will update their method, while, in this approach, we have to change our method.

In this way at least we will minimize the usage of modulestore.

Reply:
both thing are the same, in any case if edx change or update the method definition then at the end we need to update our code as well

Reply:
secondly, we are still relying on modulestore for deleting the course.